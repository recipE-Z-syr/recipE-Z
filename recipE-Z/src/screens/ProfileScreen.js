import React, { Component } from 'react';
import { Button, StyleSheet, Text, TextInput, View, SafeAreaView, ScrollView, Keyboard, TouchableOpacity, Image, Dimensions } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import defaultStyles from './stylesheet';
import NavigationBar from './NavigationBar';

class PageHeader extends React.Component {
	render () {
		const {title, trianglePosition} = this.props;
		return (
		<View style= {{position: 'absolute', top: 60}} >
		  <Image source={require('../img/header_triangle.png')} style={{width: 15, height: 10, right: trianglePosition}} />
		  <Image source={require('../img/header_rectangle.png')} style={{width: Dimensions.get('window').width, height: 60}} />
		  <Text style={{fontFamily: 'nunito-black', color: '#fff', fontSize: 28, top: -50, left: 25}}>{title}</Text>
		</View>
		);
	}
}


class ProfileScreen extends React.Component {
  render() {
    const {navigation} = this.props;
    return (
      <View style={[defaultStyles.container, {backgroundColor:'white'}]}>
      <NavigationBar navigation={this.props.navigation}/>
      <PageHeader title='Settings' trianglePosition={-364}/>
			<View style={{alignItems: 'flex-start', justifyContent: 'flex-start'}}>
				<Image source={require('../img/profilePicture.png')} style={{alignSelf: 'center', marginBottom: 55, marginTop: 35}}/>
				<View style={{flexDirection: 'row', alignItems: 'center', marginBottom: 50, marginLeft: -25}}>
					<Image source={require('../img/Name.png')} style={{height: 20, width: 20, marginRight: 35}}/>
					<Text style={[defaultStyles.h1, {fontSize: 18}]}>John Doe</Text>
				</View>
				<View style={{flexDirection: 'row', alignItems: 'center', marginBottom: 50, marginLeft: -25}}>
					<Image source={require('../img/Location.png')} style={{height: 20, width: 20, marginRight: 35}}/>
					<Text style={[defaultStyles.h1, {fontSize: 18}]}>San Francisco, CA</Text>
				</View>
				<View style={{flexDirection: 'row', alignItems: 'center', marginBottom: 50, marginLeft: -25}}>
					<Image source={require('../img/email.png')} style={{height: 20, width: 20, marginRight: 35}}/>
					<Text style={[defaultStyles.h1, {fontSize: 18}]}>john.doe@mail.com</Text>
				</View>
				<View style={{flexDirection: 'row', alignItems: 'center', marginBottom: 50, marginLeft: -25}}>
					<Image source={require('../img/password.png')} style={{height: 20, width: 20, marginRight: 35, marginBottom: 10}}/>
					<Text style={[defaultStyles.h1, {fontSize: 18}]}>●●●●●●●</Text>
				</View>
			<TouchableOpacity
        style={[defaultStyles.settingsButton]}
        onPress={() => navigation.navigate('Search')}
        underlayColor='#ed4848'>
        <Text style={defaultStyles.redButtonText}>SAVE</Text>
      </TouchableOpacity>
			</View>
			</View>
    ); //end return

 }//end render
}//end class

export default ProfileScreen;

const styles = StyleSheet.create({
});
