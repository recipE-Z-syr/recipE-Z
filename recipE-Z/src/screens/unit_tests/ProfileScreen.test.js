//ProfileScreen.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import ProfileScreen from '../ProfileScreen';

describe('<ProfileScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProfileScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('Navigates to Home', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<ProfileScreen {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(0)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Search')
})
