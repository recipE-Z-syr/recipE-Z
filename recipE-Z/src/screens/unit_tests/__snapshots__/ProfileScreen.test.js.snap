// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "fontFamily": "nunito-regular",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row-reverse",
        "position": "absolute",
        "right": 30,
        "top": 30,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
        "top": 60,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 10,
          "right": -364,
          "width": 15,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 750,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "nunito-black",
          "fontSize": 28,
          "left": 25,
          "top": -50,
        }
      }
    >
      Settings
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 55,
          "marginTop": 35,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 50,
          "marginLeft": -25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 35,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-extrabold",
              "fontSize": 20,
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        John Doe
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 50,
          "marginLeft": -25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 35,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-extrabold",
              "fontSize": 20,
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        San Francisco, CA
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 50,
          "marginLeft": -25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 35,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-extrabold",
              "fontSize": 20,
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        john.doe@mail.com
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 50,
          "marginLeft": -25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginBottom": 10,
            "marginRight": 35,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-extrabold",
              "fontSize": 20,
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        ●●●●●●●
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ed4848",
          "borderColor": "#ed4848",
          "borderRadius": 5,
          "borderWidth": 1,
          "bottom": -40,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
          "width": 250,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "nunito-bold",
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }
      >
        SAVE
      </Text>
    </View>
  </View>
</View>
`;
